### PR TITLE
[ALLUXIO-2067] Clean up leaked threads

### PR DESCRIPTION
**Pull request by aaudiber**
https://alluxio.atlassian.net/browse/ALLUXIO-2067

Before this change, running the integration tests built up around 1000 threads. After the change, it is closer to 500.

### CODE REVIEW

File: tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java [hunk 1]
```diff
@@ -97,21 +97,22 @@ public void run() {
...
   private static final long TEST_TIME_MS = Constants.SECOND_MS;
 
-  private ClientThread mCreateFileThread = null;
+  private ClientThread mCreateFileThread;
   /** Executor for running client threads. */
-  private final ExecutorService mExecutorsForClient = Executors.newFixedThreadPool(1);
-  private Configuration mMasterConfiguration = null;
+  private ExecutorService mExecutorsForClient;
+
+  public SystemPropertyRule mDisableHdfsCacheRule =
```
Comment (apc999) [inline tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java]: is this the way we are supposed to use `SystemPropertyRule`?
shall we declare it as `@ClassRule`?

File: tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java [hunk 2]
```diff
@@ -97,21 +97,22 @@ public void run() {
...
   public final void after() throws Exception {
     mExecutorsForClient.shutdown();
-    System.clearProperty("fs.hdfs.impl.disable.cache");
     ConfigurationTestUtils.resetConfiguration();
   }
 
   @Before
   public final void before() throws Exception {
-    System.setProperty("fs.hdfs.impl.disable.cache", "true");
+    mExecutorsForClient = Executors.newFixedThreadPool(1);
```
Comment (apc999) [inline tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java]: is this change related to thread-leaking? what is the difference between creating a threadpool per test case vs. per class?

**Comment by apc999** [comment]
Thanks for the fix . It looks good to me almost. Left two questions.

File: tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java (hunk 2 above)
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java]: I was auditing our threadpools and noticed that this was had creation/cleanup asymmetry. I don't think the changes to this file fix any thread-leaking.

File: tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java (hunk 1 above)
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java]: Good catch this should be ClassRule

**Comment by aaudiber** [comment]
Thanks for the comments, PTAL at the updates

**Comment by apc999** [comment]
+1 LGTM. thanks for the fix
